### ISSUE TITLE
to run this application you must install .net core

### ISSUE THREAD

**Issue by dontaskdonttellme** [micahmo/WgServerforWindows#116]
Windows 11 latest patches.  installed both .net 6 runtime and .net 7 runtime.  When I try to run the application I get "to run this application you must install .net core".   Every other program works fine on my computer.  What is going on?  yes, my account has full admin priviliges

**Comment by micahmo** [comment]
Hi @heatsiphon, sorry to hear about this issue! Let me clarify a couple things first.

1. This application currently targets .NET Core 3.1, which is why installing 6 and 7 didn't help.
2. However, the installer is supposed to automatically install the necessary dependencies (i.e., the 3.1 runtime).

Did you try to install the latest version (2.0.10)? If so, did you receive the following prompt and press install here? And if so, did that seem to succeed?

![image](https://github.com/micahmo/WgServerforWindows/assets/7417301/5d49062f-3a21-4cc1-9718-a454134f0457)

I just tested the installer on a clean Win 11 with all updates (as of today). The installer worked and the app ran fine.

**Comment by heatsiphon** [comment]
Hi Micah,

 

Thanks for your reply.

 

I just tried installing .net 3.1.32 runtime x64 on my Windows 11 (fully updated) and it still gives me the same error message.


When I click the download it now button, nothing happens at all.  I am not having an issue with any other program on my windows 11 pc. (I am an IT guy and very versed in W11).

 

I still can’t get the program to run, because I can’t get past this message.

 

I am running my W11 pro in full administrator mode.  

 

 



 

Here are my installed runtimes.

**Comment by micahmo** [comment]
@heatsiphon, you didn't answer my question about whether the WS4W installer prompted you to install .NET Core, and if so, whether it worked? If you're intentionally trying to install it manually, that's 100% fine, I'm just curious because this is supposed to all be automatic.

> When I click the download it now button, nothing happens at all.

What are you downloading?

> Here are my installed runtimes.

If there was supposed to be a screenshot here, I don't think it came through. 😞

**Comment by heatsiphon** [comment]
When the dialog box comes up and tells me I don’t have the correct version of .net installed it gives me 2 choices.  Install the missing .net core, yes or no.

 

If I click no, I get dropped back out to the desktop.

 

If I click Yes (install it), nothing happens at all.

 

The screenshots were of these boxes and a list of the .NET runtimes on my system.

**Comment by micahmo** [comment]
@heatsiphon You still have not answered my original question. 😊 Maybe I'm asking the wrong question, and you're welcome to tell me that, but otherwise it's a key piece of information that I'm missing. The WS4W installer should prompt you to install .NET Core 3.1. Did it do so? Did it work?

If you don't want to use the installer or you don't want to install .NET Core this way, that's totally fine. I just need to know that to help you troubleshoot.

**Comment by heatsiphon** [comment]
Hello –

 

I have tried to answer your question several times---

 

The dialog box comes up and says “to run this application, you must install .net core.”  Would you like to download it now.

 

If I click YES, nothing happens.


if I click NO, the dialog box closes and the program quits.

 

I have installed .net core 6 x64, .net core 7 x64, .net core 3.1.32 x64 and x86

 

Doesn’t make a difference, I still can’t launch the program.

 

Read this link:

https://stackoverflow.com/questions/59367037/published-net-core-app-warns-to-install-net-core-but-its-already-installed

 

look at your Program’s PUBLISH settings.

 

Make sure target runtime is x64

 

 

Also make sure MY_PROGRAM_NAME.runtimeconfig.json is included in the installer package

**Comment by micahmo** [comment]
When the dialog box appears, what are you launching? The installer (named something like `WS4WSetup-2.0.10.exe`) or the application itself (`WgServerforWindows.exe`)?

**Comment by heatsiphon** [comment]
Wg Server for Windows which is 

 

"C:\Program Files (x86)\WS4W\WgServerforWindows.exe"

 

I think that is the problem – what I sent you.

This looks like it was compiled for x86 because it installs into the x86 directory, not targeted / compiled for x64

**Comment by micahmo** [comment]
Thanks, I see you're launching the app itself, which is what I suspected. It is built use the AnyCPU target, which can work on 32- or 64-bit operating systems. It prefers 64-bit. The placement in `Program Files (x86)` is for legacy reasons and is very common. (Many apps you launch from that directory are 64-bit.)

![image](https://github.com/micahmo/WgServerforWindows/assets/7417301/96a5f773-fd8e-4577-b946-f4bb55ea52f2)

You also confirmed that still haven't answered my original question about the installer (`WS4WSetup-2.0.10.exe`). When you launch that executable to install the application, it is supposed to prompt you to install .NET Core 3.1. Have you run the installer? Does it give you that prompt? Does it appear to work?

**Comment by heatsiphon** [comment]
This is the installer that I am running

 

WS4WSetup-2.0.10.exe

 

I ran the installer and it completed.


When I run the application, that’s where it asks me the .net core question.  Not the installer.

 

At no time when I click “YES” to install .net core does anything happen or open up on my computer. it just quits.

**Comment by micahmo** [comment]
Hi @heatsiphon, I recently experienced this issue with another app, so I have a new suggestion. Can you check Event Viewer > Windows Logs > Application (after trying to start WS4W)? In my case, it said it there was an issue loading a DLL for .NET 7.0.5 (`C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\7.0.5`). However, that folder was empty for me! I deleted it, and the app reverted to a previous, good install of .NET 7.0.2 and ran fine.

Long story short, that error message is not always indicative that the correct runtime isn't installed, but maybe that another corrupt one is.

**Comment by cbailes** [comment]
This issue is also present on one of my systems. Installing both .NET Core 3.1 and the latest SDK have had no effect. 'Yes' in response to the automatic download prompt just exits the application.

**Comment by micahmo** [comment]
@cbailes, sorry to hear that! Can you try the suggestion in my comment prior to yours (checking Event Viewer) and let me know if you find anything interesting?

**Comment by cbailes** [comment]
@micahmo Thanks for the quick reply. Here's the system log entry:

```
Description: A .NET Core application failed.
Application: WgServerforWindows.exe
Path: C:\Program Files (x86)\WS4W\WgServerforWindows.exe
Message: You must install or update .NET to run this application.

App: C:\Program Files (x86)\WS4W\WgServerforWindows.exe
Architecture: x64
Framework: 'Microsoft.WindowsDesktop.App', version '3.1.0' (x64)
.NET location: C:\Program Files\dotnet\

The following frameworks were found:
  5.0.17 at [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  6.0.6 at [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  7.0.8 at [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

Installing the [.NET desktop 3.1.32 runtime](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-desktop-3.1.32-windows-x64-installer) appears to have fixed it.

**Comment by micahmo** [comment]
Hey @cbailes, thanks for the update! Based on the error message, it does seem like .NET Core 3.1 was not installed prior to your manual step. I'm glad the manual step worked, but I'm not sure why it didn't get installed by the setup wizard. Just out of curiosity, did the setup run fine? Do you recall if it prompted to install .NET at all?

Anyway, thanks again, and hopefully this can help others.

**Comment by cbailes** [comment]
Everything worked perfect until the app was launched post-install, after which the option to get .NET pops up. Answering either Yes or No has the same result: the app is immediately closed.

It would probably be ideal to implement a different or redundant method of installing this dependency, but yeah, hopefully this helps!

**Comment by JackMBurch** [comment]
Agreed, or at least state which .NET version to install.

**Comment by micahmo** [comment]
Hey all! I haven't heard any recent complaints about this issue. In addition, I've upgraded to the latest LTS version of .NET 8 in version 2.1.0. On top of all that, this issue only seems to occur when there is something outside the control of WS4W that has corrupted the system (or at least what the system believes about the state of .NET). With all of that said, I am going to close this as resolved. However, if anyone has any other issues with a popup about .NET, please chime in here!
